src/zip.test.ts

```diff
@@ -82,3 +82,133 @@ describe('data second typings', () => {
     assertType<Array<[string | number, string | number]>>(actual);
   });
 });
+
+describe('strict typings', () => {
+  it('on empty tuples', () => {
+    const array: [] = [];
+    const result = zip.strict(array, array);
+    expectTypeOf(result).toEqualTypeOf<[]>();
+  });
+
+  it('on empty readonly tuples', () => {
+    const array: readonly [] = [];
+    const result = zip.strict(array, array);
+    expectTypeOf(result).toEqualTypeOf<[]>();
+  });
+
+  it('on arrays', () => {
+    const array: Array<number> = [];
+    const result = zip.strict(array, array);
+    expectTypeOf(result).toEqualTypeOf<Array<[number, number]>>();
+  });
+
+  it('on mixed typeds array', () => {
+    const array1: Array<number> = [];
+    const array2: Array<string> = [];
+    const result = zip.strict(array1, array2);
+    expectTypeOf(result).toEqualTypeOf<Array<[number, string]>>();
+  });
+
+  it('on readonly arrays', () => {
+    const array: ReadonlyArray<number> = [];
+    const result = zip.strict(array, array);
+    expectTypeOf(result).toEqualTypeOf<Array<[number, number]>>();
+  });
+
+  it('on tuples', () => {
+    const array1: [1, 2, 3] = [1, 2, 3];
+    const array2: [4, 5, 6] = [4, 5, 6];
+    const result = zip.strict(array1, array2);
+    expectTypeOf(result).toEqualTypeOf<[[1, 4], [2, 5], [3, 6]]>();
+  });
+
+  it('on readonly tuples', () => {
+    const array1: readonly [1, 2, 3] = [1, 2, 3];
+    const array2: readonly [4, 5, 6] = [4, 5, 6];
+    const result = zip.strict(array1, array2);
+    expectTypeOf(result).toEqualTypeOf<[[1, 4], [2, 5], [3, 6]]>();
+  });
+
+  it('on tuples of different lengths', () => {
+    const array1: [1, 2, 3] = [1, 2, 3];
+    const array2: [4, 5] = [4, 5];
+    const result1 = zip.strict(array1, array2);
+    expectTypeOf(result1).toEqualTypeOf<[[1, 4], [2, 5]]>();
+    const result2 = zip.strict(array2, array1);
+    expectTypeOf(result2).toEqualTypeOf<[[4, 1], [5, 2]]>();
+  });
+
+  test('on variadic tuples', () => {
+    const firstVariadic: [number, ...Array<string>] = [1, 'b', 'c'];
+    const secondVariadic: [string, ...Array<number>] = ['a', 2, 3];
+    const result = zip.strict(firstVariadic, secondVariadic);
+    expectTypeOf(result).toEqualTypeOf<
+      [[number, string], ...Array<[string, number]>]
+    >;
+  });
+});
+
+describe('data second strict typings', () => {
```

Comment:
Generally, for all dataLast tests here I think the better test would be via a pipe, we want to make sure that the typing is inferred properly in a pipe as that is the common use case for dataLast. It's unlikely that the function would be used via `zip.strict(arr2)(arr1)` or `const f = zip.strict(arr2); f(arr1);`

The way types are inferred in pipes makes some of the typing that we use for dataLast not work well and we had to revert changes in the past.

Can you change the tests so that we test them via a pipe, e.g.
```ts
   const result = pipe(
     arr1,
     zip.strict(arr2),
   );
```